src/reanimated2/layoutReanimation/web/componentUtils.ts

```diff
@@ -222,6 +218,8 @@ export function handleExitingAnimation(
   const dummy = element.cloneNode() as ReanimatedHTMLElement;
   dummy.reanimatedDummy = true;
 
+  const snapshot = element.getBoundingClientRect();
+
   element.style.animationName = '';
   // We hide current element so only its copy with proper animation will be displayed
   element.style.visibility = 'hidden';
```

Comment:
This is the important thing - we have to capture boundingClientRect before hiding element. After hiding element with visibility:hidden, browser won't calculate proper boundingClientRect. For some reason element hidden this way acts as a zero-size node.